src/libs/merge/kdbmerge.c

```diff
@@ -0,0 +1,527 @@
+
+int getNonOverlapAllExistConflicts (void)
+{
+	if (nonOverlapAllExistCounter % 3 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 3 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return nonOverlapAllExistCounter / 3;
+}
+
+int getBaseEmptyConflicts (void)
+{
+	if (nonOverlapBaseEmptyCounter % 2 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 2 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return nonOverlapBaseEmptyCounter / 2;
+}
+
+int getOverlap3different (void)
+{
+	if (overlap3different % 3 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 3 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return overlap3different / 3;
+}
+
+int getOverlap1empty (void)
+{
+	if (overlap1empty % 2 != 0)
+	{
+		fprintf (stderr, "This should be a multiple of 2 at the end of each checkSingleSet\n");
+		return -1;
+	}
+	return overlap1empty / 2;
+}
+
+int getTotalOverlaps (void)
+{
+	return getOverlap1empty () + getOverlap3different ();
+}
+
+int getTotalNonOverlaps (void)
+{
+	return getBaseEmptyConflicts () + getNonOverlapAllExistConflicts () + nonOverlapOnlyBaseCounter;
+}
+
+int getTotalConflicts (void)
+{
+	return getTotalNonOverlaps () + getTotalOverlaps ();
+}
+
+void printKs (KeySet * ks)
+{
+	Key * cur = 0;
+	fprintf (stdout, "DEBUG: Iterate over all keys:\n");
+	ksRewind (ks);
+	while ((cur = ksNext (ks)) != 0)
+	{ /* Iterates over all keys and prints their name */
+		fprintf (stdout, "DEBUG: --%s\n", keyName (cur));
+	}
+}
+
+char * strremove (char * string, const char * sub)
+{
+	size_t length = strlen (sub);
+	if (length > 0)
+	{
+		char * p = string;
+		while ((p = strstr (p, sub)) != NULL)
+		{
+			memmove (p, p + length, strlen (p + length) + 1);
+		}
+	}
+	return string;
+}
+
+/**
+ * This is in contrast to the removeRoots function
+ *  Returns -1 on error, 0 on success
+ */
+int prependStringToAllKeyNames (KeySet * result, KeySet * input, const char * string)
+{
+	if (input == NULL)
+	{
+		fprintf (stderr, "input must not be null\n");
+		return -1;
+	}
+	if (result == NULL)
+	{
+		fprintf (stderr, "result must not be null!\n");
+		return -1;
+	}
+	if (string == NULL)
+	{
+		fprintf (stderr, "string must not be null!\n");
+		return -1;
+	}
+	Key * key;
+	ksRewind (input);
+	while ((key = ksNext (input)) != NULL)
+	{
+		char * newName = elektraMalloc (strlen (keyName (key)) + strlen (string) + 1);
+		strcpy (newName, string);
+		strcat (newName, keyName (key));
+		Key * duplicateKey = keyDup (key); // keySetName returns -1 if key was inserted to a keyset before
+		int status = keySetName (duplicateKey, newName);
+		elektraFree (newName);
+		if (status < 0)
+		{
+			fprintf (stderr, "Could not set new key name!\n");
+			return -1;
+		}
+		status = ksAppendKey (result, duplicateKey);
+		if (status < 0)
+		{
+			fprintf (stderr, "Could not append key!\n");
+			return -1;
+		}
+	}
+	return 0;
+}
+
+static KeySet * removeRoots (KeySet * original, Key * root)
```

Comment:
Why plural? It removes one root?

Reply:
My idea was that each key had the root in it, but I think you are right.